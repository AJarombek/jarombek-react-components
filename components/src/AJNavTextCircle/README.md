### Overview

Navigation circle component with text describing where the button navigates to.

**Use cases:**

```javascript
import { AJNavTextCircle } from 'jarombek-react-components';

const CustomComponent = () => 
  <AJNavTextCircle
    text="Previous Page"
    textSide="left"
    direction="down"
    onClick={f=>f} />
```

### Files

| Filename                   | Description                                                                     |
|----------------------------|---------------------------------------------------------------------------------|
| `AJNavTextCircle.js`       | The reusable `AJNavTextCircle` component.                                       |
| `index.js`                 | Index file which exports the `AJNavTextCircle` component.                       |
| `styles.js`                | JSS styles for the `AJNavTextCircle` component.                                 |
