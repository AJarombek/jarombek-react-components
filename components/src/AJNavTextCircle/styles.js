/**
 * Stylesheet for the AJNavTextCircle component using JSS.
 * Author: Andrew Jarombek
 * Date: 12/29/2019
 */

export default {
  ajNavTextCircle: {
    display: 'flex',
    alignItems: 'center',
  },
  textRight: {
    flexDirection: 'row',
  },
  textLeft: {
    flexDirection: 'row-reverse',
  },
  description: {
    margin: '0 10px',
  },
};
