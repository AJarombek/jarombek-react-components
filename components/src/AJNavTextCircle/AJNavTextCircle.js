/**
 * Navigation circle component with text describing where the button navigates to.
 * @author Andrew Jarombek
 * @since 12/29/2019
 */

import React from 'react';
import PropTypes from 'prop-types';
import {createUseStyles} from 'react-jss';

import styles from './styles';

const useStyles = createUseStyles(styles);

const AJNavTextCircle = () => {
  const classes = useStyles();

  return (
    <div>
    </div>
  );
};

AJNavTextCircle.propTypes = {

};

export default AJNavTextCircle;
