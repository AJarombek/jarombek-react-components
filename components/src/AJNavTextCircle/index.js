import AJNavTextCircle from './AJNavTextCircle';

export default AJNavTextCircle;
