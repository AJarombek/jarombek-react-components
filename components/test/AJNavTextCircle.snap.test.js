import React from 'react';
import AJNavTextCircle from '../src/AJNavTextCircle/AJNavTextCircle';
import renderer from 'react-test-renderer';

/**
 * Snapshot test for the {@link ../../components/AJNavTextCircle} component.
 * @author Andrew Jarombek
 * @since 12/29/2019
 */

it('renders correctly', () => {
  const tree = renderer
    .create(
      <AJNavTextCircle
        text="Navigate"
        textSide="left"
        height="40px"
        active={true}
        direction="right"
        onClick={(f) => f}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
