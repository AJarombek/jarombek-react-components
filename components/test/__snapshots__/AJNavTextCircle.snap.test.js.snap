// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ajNavTextCircle-0-2-1 textLeft-0-2-3"
>
  <div
    className="aj-nav-circle ajNavCircle-0-2-5 ajNavCircle-d0-0-2-13 active-0-2-6"
    onClick={[Function]}
  >
    <figure>
      <img
        alt=""
        className="nav-right right-0-2-12 img-0-2-8 img-d1-0-2-14"
        src="test-file-stub"
      />
    </figure>
  </div>
  <div
    className="description-0-2-4"
  >
    Navigate
  </div>
</div>
`;
